core/src/main/java/bisq/core/dao/state/model/DaoState.java

```diff
@@ -176,6 +189,10 @@ public Message toProtoMessage() {
     }
 
     public static DaoState fromProto(protobuf.DaoState proto) {
+        Map<String, Tx> txMap = proto.getBlocksList().stream()
```

Comment:
The cache initialization code doesn't need to live in the protobuf handlers. The cache is an internal optimization of the `DaoState`. By passing it in it isn't clear if that state could also live elsewhere and expands the testing breadth.

Reply:
Yes, perhaps it would be slightly nicer to move the txMap initialisation logic into the private DaoState constructor. (It makes no difference either way to anything outside this class.)

The protobuf.DaoState object doesn't include the txMap field, BTW, so it's not being passed in here.

Reply:
If I wanted to write a test and instantiate a `DaoState` I would need to duplicate logic in order to create the cache and pass it in for the behavior to be correct. I realize we don't have those tests right now, but moving towards a structure that enables testing instead of making it more complicated may get people to actually do it.